vignettes/data-in-teal.Rmd

```diff
@@ -0,0 +1,248 @@
+---
+title: "Including Data in teal Applications"
+author: "NEST CoreDev"
+output: 
+  bookdown::html_document2:
+    base_format: rmarkdown::html_vignette
+    toc: true
+    toc_depth: 5
+    number_sections: false
+vignette: >
+  %\VignetteIndexEntry{Including Data in teal Applications}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+## Data in `teal` Applications
+
+While the `teal` framework is mainly geared towards clinical data that conforms to the `ADaM` standard, general, non-relational data is handled just as well and the mechanism of passing data to applications is virtually the same.
+Also, modules in `teal.modules.general` have been designed to work with general data.
+
+All applications use the `teal_data` class as a data container.
+`teal_data` objects are passed to `init` to build the application, where modified by the filter panel (if applicable) and passed on to modules.
+Thus, the first step of building `teal` app is creating a `teal_data` object.
+
+### General Data
+
+A `teal_data` object is creating the `teal_data` function and passing data objects as `name:value` pairs.
+
+```{r, message=FALSE}
+library(teal)
+
+# create data objects
+IRIS <- iris
+CARS <- mtcars
+
+# create teal_data
+data <- teal_data(IRIS = IRIS, CARS = CARS)
+```
+
+Note that `IRIS` and `CARS` have been added to the `datanames` property of `data` (jump to [`teal_data` properties]).
+
+This is sufficient to run a `teal` app.
+
+```{r}
+# build app
+app <- init(
+  data = data,
+  modules = example_module()
+)
+
+# run app
+if (interactive()) shinyApp(app$ui, app$server)
+```
+
+### ADaM Data
+
+The container used for `CDISC` data is the `cdisc_data` class, which is an extension of `teal_data` created with the `cdisc_data` function.
+The `ADaM` data model allows us to take some short cuts when creating the object.
+For example, relationships between datasets are known and so in most cases default join keys can be set (jump to [`teal_data` properties]).
+See the documentation for `cdisc_data` for details.
+
+In the example below, two standard `CDISC` datasets (`ADSL` and `ADTTE`) are passed to `cdisc_data`.
+
+```{r}
+adsl <- data.frame(
+  STUDYID = "study",
+  USUBJID = 1:10,
+  SEX = sample(c("F", "M"), 10, replace = TRUE),
+  AGE = rpois(10, 40)
+)
+adtte <- rbind(adsl, adsl, adsl)
+adtte$PARAMCD <- rep(c("OS", "EFS", "PFS"), each = 10)
+adtte$AVAL <- c(
+  rnorm(10, mean = 700, sd = 200), # dummy OS level
+  rnorm(10, mean = 400, sd = 100), # dummy EFS level
+  rnorm(10, mean = 450, sd = 200) # dummy PFS level
+)
+
+data_cdisc <- cdisc_data(ADSL = adsl, ADTTE = adtte)
```

Comment:
Let's use `teal.data` dummy data to simplify an example

```suggestion
data_cdisc <- cdisc_data(ADSL = teal.data::rADSL, ADTTE = teal.data::rADSL)
```

Reply:
Perfect!